src/components/Card.js

```diff
@@ -20,17 +42,25 @@ export class Card {
     event.target.closest(".card").remove();
   }
 
-  _likeCard(event) {
-    event.target.classList.toggle("card__button-clicked");
+  _likeCard() {
+    this._likeButton.classList.toggle("card__button-clicked");
   }
 
   _setEventListeners() {
     this._templateClone
       .querySelector(".card__button")
-      .addEventListener("click", this._likeCard);
+      .addEventListener("click", () => {
+        this.handleLike(this.id, this.isLiked)
+          .then(this._likeCard())
+          .then((this.isLiked = !this.isLiked))
```

Comment:
In `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well.